### DISCUSSION TITLE
Improving the documentation: what's next?

### DISCUSSION THREAD

**Discussion by GeorgesStavracas** [flatpak/xdg-desktop-portal#1232] [Ideas]
Over the past month and a half, our documentation and website received major facelifts, and look wonderful now! Have a look:

![image](https://github.com/flatpak/xdg-desktop-portal/assets/3518204/f53e4a4b-7ed2-4e7e-9f38-5dacc0be2a6f)

![image](https://github.com/flatpak/xdg-desktop-portal/assets/3518204/f899812b-036c-43d7-84ce-6ca204860449)

However, this begs the question: is the current documentation enough? Is the website enough?

To me, the answer to both questions is a solid "probably not". But I also don't personally have much of an idea of what makes documentation good, what kind of contents people are looking for, *who* is even looking at our documentation, etc. Which is why I'm opening this up for broader discussion.

Some ideas that come to my mind:

 * Document how to write a new portal backend from scratch
 * Split and simplify the "Common Conventions" section
 * Document what configuration knobs XDG Desktop Portal offers ("Portals for system administrators"?)
 * Make gdbus-codegen generate better titles for documentation pages (?)
 * Remove a lot of of the website's content, leave only the documentation button and a call to action (?)

I'd appreciate to learn if anyone has other ideas on how to improve that!

**Comment by ebassi** [comment]
The main missing bit is probably a proper documentation on the expectations of an implementation of a portal:

- what are the keys in the `.portal` file, and how do they work
- what are the dependencies between portal implementations, for instance: if I want to provide an implementation for the FileChooser portal, what other portals do I need to implement?

Additionally, I'd like to documentation on the internals of the code base, for instance: how does the document portal work?

**Comment by bilelmoussaoui** [comment]
I would also go ahead and document the tools available for debugging the portals: Doorknocker & ashpd demo comes to mind. Maybe also mention the available client side libraries

**Reply by whot** [reply]
fwiw,  I wrote https://gitlab.freedesktop.org/whot/portal-poker/ while doing the inputcapture work though it's (mostly) sitting abandoned right now

**Comment by hfiguiere** [comment]
Contributors guide
- how do I contribute
- setting up to run the test (not trivial I found)

Internals
- How things work internally.

**Comment by GeorgesStavracas** [comment]
https://github.com/flatpak/xdg-desktop-portal/pull/1391 should get us closer to what has been suggested. Further improvements are always welcomed, of course!
